phoenix-core/src/main/java/org/apache/phoenix/schema/PTableKey.java

```diff
@@ -26,7 +28,8 @@
     public PTableKey(PName tenantId, String name) {
         Preconditions.checkNotNull(name);
         this.tenantId = tenantId;
-        this.name = name;
+        this.name = !name.contains(QueryConstants.NAMESPACE_SEPARATOR) ? name
```

Comment:
So if the table name was A:B, the key would become A.B. Can you add some test case around this and make sure creating, query, upserting to a table named like this works fine.